### PR TITLE
Fix dependecy updates

### PR DESCRIPTION
**Pull request by AleksMat**
This should fix all remaining failing tests and pylint issues. Note that CI will fail because it need new versions of sentinelhub-py and s2cloudless.

### CODE REVIEW

File: io/eolearn/io/local_io.py [hunk 1]
```diff
@@ -294,11 +294,11 @@ def execute(self, eopatch, *, filename=None):
 
         channel_count, height, width = image_array.shape
 
-        src_crs = eopatch.bbox.crs.epsg
+        src_crs = {'init': eopatch.bbox.crs.ogc_string()}
```
Comment (batic) [inline io/eolearn/io/local_io.py:297]: don't use 'init' here, to get rid of (at least some) warnings

File: io/eolearn/io/local_io.py [hunk 2]
```diff
@@ -294,11 +294,11 @@ def execute(self, eopatch, *, filename=None):
...
         src_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
 
         if self.crs:
-            dst_crs = self.crs.epsg
+            dst_crs = {'init': self.crs.ogc_string()}
```
Comment (batic) [inline io/eolearn/io/local_io.py:301]: same here

File: mask/eolearn/mask/cloud_mask.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class AddCloudMaskTask(EOTask):
     """
     def __init__(self, classifier, data_feature, cm_size_x=None, cm_size_y=None, cmask_feature='CLM',
                  cprobs_feature=None, instance_id=None, data_collection=None,
-                 image_format=MimeType.TIFF, model_evalscript=MODEL_EVALSCRIPT, data_source=None):
+                 image_format=MimeType.TIFF, model_evalscript=None, data_source=None):
```
Comment (batic) [inline mask/eolearn/mask/cloud_mask.py:56]: We should remove these parameters (`image_format`, `model_evalscript`, `data_source`, `instance_id`, `data_collection`)? Either already available data (in eopatch) or (official) evalscript from the s2cloudless to download the data should be used. 

_If not now, when s2cloudless is updated..._

**Review by batic — COMMENTED** [review]
Just a few notes. Otherwise looks good.

File: io/eolearn/io/local_io.py (hunk 1 above)
Comment (AleksMat) [reply io/eolearn/io/local_io.py:297]: Without this it doesn't work. Using epsg code failed unit tests. I suggest that this is resolved in the next release.

File: io/eolearn/io/local_io.py (hunk 2 above)
Comment (AleksMat) [reply io/eolearn/io/local_io.py:301]: Same answer

File: mask/eolearn/mask/cloud_mask.py (hunk 3 above)
Comment (AleksMat) [reply mask/eolearn/mask/cloud_mask.py:56]: Yes, in the next release we will significantly refactor this.

File: io/eolearn/io/local_io.py (hunk 1 above)
Comment (batic) [reply io/eolearn/io/local_io.py:297]: Huh. Something worked for me, but I am now completely lost. I agree - let's take time to understand the changes going to rasterio 1.2...

**Pull request merged**